### PR TITLE
Dispersion optics

### PR DESCRIPTION
**Pull request by syhaffert**
I have added some optics that introduce dispersion (prism and grating) and some helper functions. These will be useful for simulating spectrographs and ADCs. I have tried to make comprehensive tests and document all functionality.

### CODE REVIEW

**Review by ehpor — CHANGES_REQUESTED** [review]
Overall looks like a worthwhile change and definitely something that I'll use myself instead of the few rudimentary classes that I built personally and never bothered to clean up.

Some PEP8 stylistic changes regarding assignment of lambdas to variables. These should be replaced by a function definition. There are a few more than I indicated, so look yourself at other places where you made the same mistake. Also some missing units in the docstrings (also way more than I indicated; just look through all the places where there is an angle and add "radians").

**Comment by syhaffert** [comment]
I have addressed your comments and made some changes. I think it should be ready now.

File: hcipy/optics/dispersion_optics.py [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+		'''
+		return Field(grid.rotated(self._orientation).y * np.tan(self._angle), grid)
+
+class ThinPrism(TiltElement):
+	'''A thin prism that operates in the paraxial regime.
+
+	Parameters
+	----------
+	angle : scalar
+		The wedge angle of the prism.
```
Comment (ehpor) [inline hcipy/optics/dispersion_optics.py:131]: Still a missing unit here.

**Review by ehpor — CHANGES_REQUESTED** [review]
The rest looks good.

**Review by ehpor — APPROVED** [review]
Thanks for all these new optical elements!

**Pull request merged**